### ISSUE TITLE
Module content just stopped displaying

### ISSUE THREAD

**Issue by pyrosmiley** [cbrooker/MMM-Todoist#16]
Not really sure *what's* going on with this one. It worked excellently for awhile (several weeks at least!) and then __boom__ it's stopped showing up since the other day. The module header is there, but there's no list. I've checked my config and I've deleted/reinstalled the module entirely, but there's just no content. Thought I had maybe messed it up while trying to change CSS but I didn't modify this one at all.

**Comment by hutny1980** [comment]
Myself and a friend are experiencing the same issue as @pyrosmiley

**Comment by larsauswsw** [comment]
Maybe it has to do with the following exception:

TypeError: parser.close is not a function
0|mm       |     at SAXStream.openf1 (/home/pi/MagicMirror/node_modules/feedme/lib/xmlfeedparser.js:107:14)
0|mm       |     at emitOne (events.js:96:13)
0|mm       |     at SAXStream.emit (events.js:188:7)
0|mm       |     at Object.me._parser.(anonymous function) [as onopentag] (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:258:17)
0|mm       |     at emit (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:624:35)
0|mm       |     at emitNode (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:629:5)
0|mm       |     at openTag (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:825:5)
0|mm       |     at Object.write (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:1278:15)
0|mm       |     at SAXStream.write (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:239:18)
0|mm       |     at yoshi.write (/home/pi/MagicMirror/node_modules/feedme/lib/feedme.js:41:14)
0|mm       | Whoops! There was an uncaught exception...
0|mm       | TypeError: parser.close is not a function
0|mm       |     at SAXStream.openf1 (/home/pi/MagicMirror/node_modules/feedme/lib/xmlfeedparser.js:107:14)
0|mm       |     at emitOne (events.js:96:13)
0|mm       |     at SAXStream.emit (events.js:188:7)
0|mm       |     at Object.me._parser.(anonymous function) [as onopentag] (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:258:17)
0|mm       |     at emit (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:624:35)
0|mm       |     at emitNode (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:629:5)
0|mm       |     at openTag (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:825:5)
0|mm       |     at Object.write (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:1278:15)
0|mm       |     at SAXStream.write (/home/pi/MagicMirror/node_modules/sax/lib/sax.js:239:18)
0|mm       |     at yoshi.write (/home/pi/MagicMirror/node_modules/feedme/lib/feedme.js:41:14)

**Comment by larsauswsw** [comment]
@hutny1980 @pyrosmiley Did you made the update of the MagicMirror to version 2.13 from 2017-10-01?

**Comment by larsauswsw** [comment]
Have the same issue after/with update of MagicMirror.

**Comment by hutny1980** [comment]
@larsauswsw - yes, recently upgraded MagicMirror and still not working. Here's the config:

 {
                module: 'MMM-Todoist',
                position: 'top_left',   // This can be any of the regions. Best results in left or right regions.
                header: 'Reminders', // This is optional
                config: { // See 'Configuration options' for more information.
                accessToken: 'REMOVED',
                        maximumEntries: 60,
                        updateInterval: 60,
                        projects: [2163568809],
                        fade: false,
                        showProject: false
      }
        },

**Comment by baz4096** [comment]
Figured out the problem.

In MMM-Todoist.js

apiBase: "https://todoist.com/API/",

But it looks like Todoist are enforcing case sensitive URLs now (perhaps this is why there's an unexpected issue)

I changed mine to 

apiBase: "https://todoist.com/api",

And it works now.

I also dropping the trailing '/' due to it being doubled up when used in the fetcher.js code.

Too much of a github noob to work out how to commit changes I'm afraid, hopefully someone else can do so? :(

**Comment by hutny1980** [comment]
@baz4096 - how the heck did you work that out!! Changed apiBase to "https://todoist.com/api", as suggested and working like a charm.

Huge thanks for all your help.

**Comment by baz4096** [comment]
@hutny1980 You're very welcome. I'm an ex-web php/js dev and now a full time c++ dev, so I just stepped through the code and added some console.error("Messages") in various places until I figured it out.

I suspect that http://todoist.com/API was recently setup to redirect to their API docs, and most likely broke their actual REST API available on http://todoist.com/api/

**Comment by pyrosmiley** [comment]
@hutny1980 @baz4096 *wow* you're awesome, thanks so much!! All I know is [__very__] rudimentary python, so trying to sus out the problem was not going well. AFAIK, the only way to ask for commits is to fork it and then put in a pull request. If you don't mind, I'll go ahead and do it (with full credit, of course!!) just so that the fix is available for people. That way there's something there until the original author can get to merging everything in the main repository.

**Comment by pyrosmiley** [comment]
I made the change and submitted a pull request [right here](https://github.com/cbrooker/MMM-Todoist/pull/17) at the main branch. 

For anyone just stumbling across this issue, until that pull request is merged into the main distro, I have a working clone of the module [that I forked](https://github.com/pyrosmiley/MMM-Todoist) to integrate the fix, so if you'd rather just download a working version of it temporarily, that should do. I have no plans to continue updating it or make any other changes though, so it probably won't stay up-to-date with OP's module!
